From 6ff6edf8d7a59b852a7e6084f658a1d3e1ee7500 Mon Sep 17 00:00:00 2001
From: Raphael Kubo da Costa <rakuco@FreeBSD.org>
Date: Tue, 30 Dec 2014 14:23:49 +0200
Subject: [PATCH] Abi: Correctly set ABI information for the BSDs in hostAbi().

Add another #ifdef block for BSD operating systems so that qtcreator
stops complaining that it does not have enough information about the
host's ABI.

Change-Id: I5841e162eda7f34fa4a962ab1c035d1727613f6f
Reviewed-by: Tobias Hunger <tobias.hunger@theqtcompany.com>

--- src/plugins/projectexplorer/abi.cpp
+++ src/plugins/projectexplorer/abi.cpp
@@ -744,6 +744,16 @@ Abi Abi::hostAbi()
     os = MacOS;
     subos = GenericMacFlavor;
     format = MachOFormat;
+#elif defined (Q_OS_BSD4)
+    os = BsdOS;
+# if defined (Q_OS_FREEBSD)
+    subos = FreeBsdFlavor;
+# elif defined (Q_OS_NETBSD)
+    subos = NetBsdFlavor;
+# elif defined (Q_OS_OPENBSD)
+    subos = OpenBsdFlavor;
+# endif
+    format = ElfFormat;
 #endif
 
     const Abi result(arch, os, subos, format, QSysInfo::WordSize);

